six/modules/c++/samples/test_large_offset.cpp

```diff
@@ -0,0 +1,164 @@
+    {
+        data = six::sicd::Utilities::createFakeComplexData();
+        data->setNumRows(52440); // (11GB / 4) ^ (1/2)
+        data->setNumCols(52440);
+        data->setPixelType(six::PixelType::RE16I_IM16I);
+    }
+    else
+    {
+        data = six::sidd::Utilities::createFakeDerivedData();
+        data->setNumRows(74152); // (11GB / 2) ^ (1/2)
+        data->setNumCols(74162);
+        data->setPixelType(six::PixelType::MONO16I);
+    }
+
+    const size_t bytesPerRow = data->getNumCols() * data->getNumBytesPerPixel();
+    const size_t imageSize = data->getNumRows() * bytesPerRow;
+
+    mem::SharedPtr<six::Container> container(
+            new six::Container(data->getDataType()));
+    container->addData(data.release());
+
+    six::NITFWriteControl writer;
+    writer.setXMLControlRegistry(&registry);
+    writer.initialize(container);
+
+    mem::ScopedArray<six::UByte> imageData(new six::UByte[imageSize]);
+    for (size_t ii = 0; ii < imageSize; ++ii)
+    {
+        imageData[ii] = static_cast<six::UByte>(ii);
+    }
+    writer.save(imageData.get(), outputPathname);
+}
+
+bool checkNITF(const std::string& pathname)
+{
+    six::XMLControlRegistry registry;
+    registry.addCreator(six::DataType::COMPLEX,
+            new six::XMLControlCreatorT<
+                    six::sicd::ComplexXMLControl>());
+    registry.addCreator(six::DataType::DERIVED,
+            new six::XMLControlCreatorT<
+                    six::sidd::DerivedXMLControl>());
+
+    six::NITFReadControl reader;
+    reader.setXMLControlRegistry(&registry);
+    const std::vector<std::string> schemaPaths;
+    reader.load(pathname, schemaPaths);
+    std::auto_ptr<six::Data> data(reader.getContainer()->getData(0)->clone());
+
+    six::Region region;
+    region.setStartRow(ROWS_TO_SKIP);
+    region.setNumRows(data->getNumRows() - ROWS_TO_SKIP);
+    region.setStartCol(0);
+    reader.interleaved(region, 0);
+    six::UByte* buffer = region.getBuffer();
+
+    const size_t bytesPerRow = data->getNumCols() * data->getNumBytesPerPixel();
+    const size_t skipSize = ROWS_TO_SKIP * bytesPerRow;
+    const size_t imageSize = data->getNumRows() * bytesPerRow;
+
+    for (size_t ii = skipSize; ii < imageSize; ++ii)
+    {
+        if (buffer[ii - skipSize] != static_cast<six::UByte>(ii))
+        {
+            return false;
+        }
+    }
+    return true;
+}
+
+bool runTest(const six::DataType& datatype)
+{
+    std::cerr << "Running offset test for datatype " << datatype.toString()
+              << "\n";
+
+    const io::TempFile temp;
+    createNITF(temp.pathname(), datatype);
+    if (checkNITF(temp.pathname()))
+    {
+        std::cout << "Test passed\n";
+        return true;
+    }
+    std::cerr << "Test failed\n";
+    return false;
+}
+}
+
+int main(int argc, char** argv)
+{
+    try
+    {
+        bool testPassed = runTest(six::DataType::COMPLEX);
+        testPassed = runTest(six::DataType::DERIVED);
```

Comment:
This logic needs to be a little different... right now if SICD failed but SIDD succeeded, we'd return 0